### PR TITLE
xtb - fix `la_backend` logic

### PR DESCRIPTION
**Pull request by ltalirz**
It appears `la_backend` is to be introduced in 6.6.1. Correcting the logic to reflect this.

### CODE REVIEW

**Comment by ltalirz** [comment]
@awvwgk

**Comment by ltalirz** [comment]
I still need to test it (but I suspect it will work)

**Comment by alalazo** [comment]
Waiting for @awvwgk before merging.

**Comment by awvwgk** [comment]
Currently fails for me due to a change in the meson wrap settings.

<details>

````
[+] /home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/tblite-0.3.0-xrgu6s5jtijf5yiajher2zt2fmlzpeco
==> Installing xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf
==> No binary for xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf found: installing from source
==> Fetching https://github.com/grimme-lab/xtb/releases/download/v6.6.0/xtb-6.6.0-source.tar.xz
==> No patches needed for xtb
==> xtb: Executing phase: 'meson'
==> Error: ProcessError: Command exited with status 1:
    '/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/meson-1.1.0-cwliluiyaxqskfgwevbqcla2chnbqfjo/bin/meson' 'setup' '/tmp/REDMOND.sehlert/spack-stage/spack-stage-xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/spack-src' '-Dprefix=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf' '-Dlibdir=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/lib' '-Dbuildtype=release' '-Dstrip=false' '-Ddefault_library=shared' '-Dwrap_mode=nodownload' '-Dlapack=openblas' '-Dopenmp=true'

1 error found in build log:
     41    |   any errors please try upgrading CMake to a newer version first.
     42    
     43    WARNING: CMake Toolchain: Failed to determine CMake compilers state
     44    Run-time dependency test-drive found: NO (tried pkgconfig and cmake)
     45    Looking for a fallback subproject for the dependency test-drive
     46    
  >> 47    ../spack-src/test/unit/meson.build:19:16: ERROR: Automatic wrap-based subproject downloading is disabled
     48    
     49    A full log can be found at /tmp/REDMOND.sehlert/spack-stage/spack-stage-xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/spack-build-wjussdd/meson-logs/meson-log.txt

See build log for details:
  /tmp/REDMOND.sehlert/spack-stage/spack-stage-xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/spack-build-out.txt

==> Error: xtb-6.5.0-eflorl73xeq2w3s3seshc3rwemi552wa: ProcessError: Command exited with status 1:
    '/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/meson-1.1.0-cwliluiyaxqskfgwevbqcla2chnbqfjo/bin/meson' 'setup' '/tmp/REDMOND.sehlert/spack-stage/spack-stage-xtb-6.5.0-eflorl73xeq2w3s3seshc3rwemi552wa/spack-src' '-Dprefix=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.5.0-eflorl73xeq2w3s3seshc3rwemi552wa' '-Dlibdir=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.5.0-eflorl73xeq2w3s3seshc3rwemi552wa/lib' '-Dbuildtype=release' '-Dstrip=false' '-Ddefault_library=shared' '-Dwrap_mode=nodownload' '-Dlapack=openblas' '-Dopenmp=true'
==> Error: xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf: ProcessError: Command exited with status 1:
    '/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/meson-1.1.0-cwliluiyaxqskfgwevbqcla2chnbqfjo/bin/meson' 'setup' '/tmp/REDMOND.sehlert/spack-stage/spack-stage-xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/spack-src' '-Dprefix=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf' '-Dlibdir=/home/sehlert/projects/spack/gh/opt/spack/linux-ubuntu20.04-zen2/gcc-9.4.0/xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf/lib' '-Dbuildtype=release' '-Dstrip=false' '-Ddefault_library=shared' '-Dwrap_mode=nodownload' '-Dlapack=openblas' '-Dopenmp=true'
==> Error: xtb-6.6.0-wjussddthjny2v3wa2jk4etsxo2ng5qf: Package was not installed
==> Error: xtb-6.5.0-eflorl73xeq2w3s3seshc3rwemi552wa: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
````
</details>

I'll have a look.

**Comment by ltalirz** [comment]
@awvwgk do you happen to know what needs to be fixed in order to be able to build xtb with spack again?

**Comment by awvwgk** [comment]
Leopold, could you try applying

```diff
diff --git a/var/spack/repos/builtin/packages/test-drive/package.py b/var/spack/repos/builtin/packages/test-drive/package.py
new file mode 100644
index 0000000000..92c77f7af5
--- /dev/null
+++ b/var/spack/repos/builtin/packages/test-drive/package.py
@@ -0,0 +1,12 @@
+from spack.package import *
+
+
+class TestDrive(MesonPackage):
+    """Simple testing framework for Fortran packages"""
+
+    homepage = "https://github.com/fortran-lang/test-drive"
+    url = "https://github.com/fortran-lang/test-drive/releases/download/v0.4.0/test-drive-0.4.0.tar.xz"
+
+    maintainers("awvwgk")
+
+    version("0.4.0", "effabe5d46ea937a79f3ea8d37eea43caf38f9f1377398bad0ca02784235e54a")
diff --git a/var/spack/repos/builtin/packages/xtb/package.py b/var/spack/repos/builtin/packages/xtb/package.py
index 064ad89bff..b93d10497e 100644
--- a/var/spack/repos/builtin/packages/xtb/package.py
+++ b/var/spack/repos/builtin/packages/xtb/package.py
@@ -23,9 +23,11 @@ class Xtb(MesonPackage):
     depends_on("meson@0.57.2:", type="build")
     depends_on("mctc-lib")
     depends_on("pkgconfig", type="build")
-    depends_on("tblite", when="@6.6.0:")
+    depends_on("tblite", when="+tblite")
+    depends_on("test-drive", type="build")
 
     variant("openmp", default=True, description="Use OpenMP parallelisation")
+    variant("tblite", default=True, when="@6.6.0:", description="Use tblite for xTB calculations")
 
     def meson_args(self):
         lapack = self.spec["lapack"].libs.names[0]
@@ -34,7 +36,7 @@ def meson_args(self):
         elif lapack != "openblas":
             lapack = "netlib"
 
-        lapack_opt = "-Dlapack={0}" if self.version <= Version("6.6.0") else "-Dla_backend={0}"
+        lapack_opt = "-Dlapack={0}" if self.version >= Version("6.6.0") else "-Dla_backend={0}"
 
         return [
             lapack_opt.format(lapack),
```

**Comment by berquist** [comment]
> Leopold, could you try applying
> 
> ...

This patch (once the conflict is resolved) fixes the installation for 6.6.0.

**Comment by ltalirz** [comment]
Thanks for the fix @awvwgk and thanks for checking @berquist !

Just one question about this change:

```
-        lapack_opt = "-Dlapack={0}" if self.version <= Version("6.6.0") else "-Dla_backend={0}"
+        lapack_opt = "-Dlapack={0}" if self.version >= Version("6.6.0") else "-Dla_backend={0}"
```

This is reverting the change I made here in the PR - is this intentional, i.e. is the original selector correct after all? I thought the `la_backend` variable is to be introduced for versions >=6.6.1?

**Comment by awvwgk** [comment]
Yes, the original logic for switching from `la_backend` to the new `lapack` option is correct. I changed this in version 6.6.0.

**Comment by ltalirz** [comment]
Ok! Sorry for the misunderstanding - in that case the PR should be ready to go

Edit: Sorry for the commit mess here, working with the github web interface

File: var/spack/repos/builtin/packages/xtb/package.py [hunk 1]
```diff
@@ -23,9 +23,11 @@ class Xtb(MesonPackage):
     depends_on("meson@0.57.2:", type="build")
     depends_on("mctc-lib")
     depends_on("pkgconfig", type="build")
-    depends_on("tblite", when="@6.6.0:")
+    depends_on("tblite", when="+tblite")
+    depends_on("test-drive", type="build")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xtb/package.py:27]:
```
1. xtb: unknown package 'test-drive' in 'depends_on' directive
     in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/xtb/package.py
2. xtb: dependency on test-drive cannot be satisfied by known versions of test-drive
    happening in /home/runner/work/spack/spack/var/spack/repos/builtin/packages/xtb/package.py
```

Comment (awvwgk) [reply var/spack/repos/builtin/packages/xtb/package.py:27]: The patch I posted was not applied completely, the missing bit is

```diff
diff --git a/var/spack/repos/builtin/packages/test-drive/package.py b/var/spack/repos/builtin/packages/test-drive/package.py
new file mode 100644
index 0000000000..92c77f7af5
--- /dev/null
+++ b/var/spack/repos/builtin/packages/test-drive/package.py
@@ -0,0 +1,12 @@
+from spack.package import *
+
+
+class TestDrive(MesonPackage):
+    """Simple testing framework for Fortran packages"""
+
+    homepage = "https://github.com/fortran-lang/test-drive"
+    url = "https://github.com/fortran-lang/test-drive/releases/download/v0.4.0/test-drive-0.4.0.tar.xz"
+
+    maintainers("awvwgk")
+
+    version("0.4.0", "effabe5d46ea937a79f3ea8d37eea43caf38f9f1377398bad0ca02784235e54a")
```

Comment (ltalirz) [reply var/spack/repos/builtin/packages/xtb/package.py:27]: sorry, I really dropped the ball here with my previous edits (clearly did not grasp Sebastian's suggestion correctly)

So, this PR not only fixes the xtb build but also adds a new test-drive package.

I hope the edits are now applied correctly. Apologies again for the delay

Comment (awvwgk) [reply var/spack/repos/builtin/packages/xtb/package.py:27]: Previously the test-drive package was downloaded on-demand, however now meson is used with no download option making the build fail. Adding the package as proper dependency is the way to go then.

**Comment by alalazo** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/test-drive/package.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from spack.package import *
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/test-drive/package.py:6]: License is missing here

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by awvwgk** [comment]
@spackbot fix style

**Review by awvwgk — APPROVED** [review]
Looks good to me now

**Comment by ltalirz** [comment]
Looks like this can be merged now?

**Review by alalazo — APPROVED** [review]

**Pull request merged**